ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandStore.java

```diff
@@ -186,6 +199,25 @@ public QueuedCommandStatus enqueueCommand(
     return restoreCommands;
   }
 
+  public CompletableFuture<Void> getConsumerPositionFuture(final long offset) {
+    final CompletableFuture<Void> future = new CompletableFuture<>();
+
+    final long consumerPosition = getConsumerPosition();
+    if (consumerPosition > offset) {
+      future.complete(null);
+    } else {
+      commandOffsetFutures.add(Pair.of(future, offset));
+    }
+    return future;
+  }
+
+  // returns the next offset to be consumed
+  private long getConsumerPosition() {
+    final Collection<TopicPartition> cmdTopicPartitions = getTopicPartitionsForTopic(commandTopic);
+    // NOTE: assumes there's only one partition
```

Comment:
So at the moment this method will periodically call out to the broker to get the number of partitions when the cluster metadata ages out (I think).

Also, the number of partitions shouldn't be changing, so this is probably unnecessary. Personally, I would:

1. Drop this comment.
2. Hardcode this to `new TopicPartition(commandTopic, 1)` - in the constructor even.
3. Update `enqueueCommands` to check which partition a command has been written too and fail hard if its written to anything other that 1. (Or is it zero indexed? Can't remember...)

Reply:
Point 3 is just to cause things to fail if a user goes and changes the number of partitions in the command topic. After all, the command topic is just a topic like any other in Kafka and the user can change its settings.

TBH though, given we _always_ write to the first partition, I wouldn't worry about it.